examples/tutorial/x/blog/handlers_test.go

```diff
@@ -1,14 +1,166 @@
 package blog
 
 import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"os"
 	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/x"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	weaveCtx weave.Context
+	txs      map[string]weave.Tx
+	handlers map[string]weave.Handler
+	objects  map[string]weave.Persistent
 )
 
+func toWeaveAddress(addr string) weave.Address {
+	d, err := hex.DecodeString(addr)
+	if err != nil {
+		panic(err)
+	}
+
+	return d
+}
+
+func newContext(helpers x.TestHelpers) weave.Context {
+	ctx := context.Background()
+	ctx = weave.WithHeight(ctx, 100)
+	return ctx
+}
+
+func newAuth(helpers x.TestHelpers, ctx weave.Context) x.Authenticator {
+	auth := helpers.CtxAuth("authKey")
+	_, addr := helpers.MakeKey()
+	auth.SetConditions(ctx, addr)
+	return auth
+}
+
+func TestMain(m *testing.M) {
```

Comment:
I'm not a fan of TestMain for data initialization. Unless you need to start a background process or do some heavy lifting.

Also, this separated the data to be tested quite far from the tests being run, and using keys makes it less obvious what is being tested. I would rather the tx and handler were set up in each test